classes/use-imx-seco-firmware.bbclass

```diff
@@ -0,0 +1,32 @@
+#
+# Class to provide a possibility to re-use names of SECO Firmware
+# files, which are required by certain derivatives on i.MX8 family.
+#
+# SECO Firmware names are used by imx-boot and imx-seco recipes,
+# therefore their respective definitions are moved into a separate
+# class, which is inherited by both recipes.
+#
+# Currently, only 'mx8x' family requires SECO FW to be provided in the
+# target image, therefore as a first step - the machine override is
+# verified if it matches, and then derivative is taken.
+#
+# NOTE: SECO_FIRMWARE_NAME defaults to empty string, and are verified
+# against the family first. If a derivative in the family does not
+# have a firmware name set in this class - recipe parsing is stopped.
+# This behavior ensures that derivatives which requires SECO Firmware
+# to be present in the image file have it properly defined.
+
+SECO_FIRMWARE_NAME              ?= ""
+
+SECO_FIRMWARE_NAME_mx8qm         = "mx8qmb0-ahab-container.img"
+SECO_FIRMWARE_NAME_mx8qxp        = "mx8qxb0-ahab-container.img"
+SECO_FIRMWARE_NAME_mx8qxpc0      = "mx8qxc0-ahab-container.img"
+SECO_FIRMWARE_NAME_mx8phantomdxl = "mx8qxb0-ahab-container.img"
+SECO_FIRMWARE_NAME_mx8dxl        = "mx8dxla0-ahab-container.img"
+
+python () {
+    if "mx8x" in d.getVar('MACHINEOVERRIDES').split(":"):
```

Comment:
The firmware is valid for both 8 and 8X families.

Reply:
Does that mean that only 8M family have no SECO FW required?

In other words, would it be safe to put something like:
```python

python () {
    if "mx8m" not in d.getVar('MACHINEOVERRIDES').split(":"):
        seco_firmware = d.getVar('SECO_FIRMWARE_NAME')
        if not seco_firmware:
            raise bb.parse.SkipRecipe("This SoC requires 'SECO_FIRMWARE_NAME', define it in 'use-imx-seco-firmware' bbclass")
}

```

Reply:
Yes, that will work.

Reply:
Thanks for the update! I've tried it out and it seems to be working for derivatives of 8 and 8X families.

I would make an adjustment and push an updated version.

Reply:
Possibly like this:

```python
python () {
    seco_firmware = d.getVar('SECO_FIRMWARE_NAME')
    if not seco_firmware and "mx8m" not in d.getVar('MACHINEOVERRIDES').split(":"):
        raise bb.parse.SkipRecipe("This SoC requires 'SECO_FIRMWARE_NAME', define it in 'use-imx-seco-firmware' bbclass")
}
```